lib/importer/src/index.js

```diff
@@ -477,6 +520,124 @@ exports.Initialise = (config, router, prototypeKit) => {
+      }
+    );
+
+
+    router.post(
+      IMPORTER_ROUTE_MAP.get("importerConfiguration"),
+      (request, response) => {
+        request.session.data['internal-message'] = "Configuration changes have been saved, and the application will restart shortly"
+
+        setTimeout(function () {
+          plugin_config.persistConfig()
+        }, 5000)
+
+        response.redirect(IMPORTER_ROUTE_MAP.get("importerConfiguration"))
+      }
+    );
+
+    router.post(
+      IMPORTER_ROUTE_MAP.get("importerConfigurationPath"),
+      (request, response) => {
+        let newPath = request.body.uploadPath
+
+        if (newPath && !fs.existsSync(newPath)) {
+          request.session.data['internal-error'] = {
+            summary: "Directory does not exist",
+            error: "The specified path does not exist on the computer where the prototype kit is executing. Please choose another path or set to empty for the default value. The plugin will not operate until this error is corrected",
+            parameter: newPath
+          }
+          response.redirect(IMPORTER_ROUTE_MAP.get("importerConfiguration"))
+          return
+        }
+
+        response.redirect(IMPORTER_ROUTE_MAP.get("importerConfiguration"))
+      }
+    )
+  } else {
+    // Disable all handling for configuration pages when not in development mode
+    router.all(
+      IMPORTER_ROUTE_MAP.get("importerConfiguration"),
+      (request, response) => {
+        response.status(200)
+        response.render("plugin_config_disabled.html")
+      })
+  }
+
+
+  function render_add_field(res, summary, error) {
+    if (summary) {
+      res.status(400)
+      res.render("plugin_config_add_field.html", { error: error, error_summary: summary })
+    } else {
+      res.status(200)
+      res.render("plugin_config_add_field.html")
+    }
+  }
+
+  // Render the add a field page
+  router.get(
+    IMPORTER_ROUTE_MAP.get("importerConfigurationField"),
+    (request, response) => {
+      render_add_field(response)
+    }
+  );
+
+  // Add or delete a field based on the action supplied by the form
+  router.post(
+    IMPORTER_ROUTE_MAP.get("importerConfigurationField"),
+    (request, response) => {
+      if (!request.body.field) {
+        render_add_field(response, "Missing value", "You must provide a name for the field using only alphanumeric characters or whitespace")
+        return
+      }
+
+      if (request.body.action == "delete") {
+        plugin_config.setFields(plugin_config.fields.filter((x) => x != request.body.field))
+      } else {
+        // We don't want to add a field if it already exists (by name), and so we will
+        // just return if it is already included.
+        if (!plugin_config.fields.find((x) => x.toUpperCase() == request.body.field.toUpperCase())) {
+          plugin_config.setFields([...plugin_config.fields, request.body.field])
+        }
+      }
+
+      response.redirect(IMPORTER_ROUTE_MAP.get("importerConfiguration"));
+    }
```

Comment:
Shouldn't this also be in an `if (developmentMode)` check? For completeness at least.

Reply:
Good catch, thanks,